### ISSUE TITLE
battle speed

### ISSUE THREAD

**Issue by jiaqp** [tacticians-academy/teamfight-simulator#208]
hi，
How to set the combat speed to achieve the purpose of obtaining the combat results in an instant
thank you

**Comment by ky-is** [comment]
Hi, good question! Do you have the project running locally on your machine? If so, there's a trivial code change you can make to instantly play out fights. There are quite a few gotchas still though, so I'll go into some detail:

We can find the game's run loop in [src/game/loop.ts](https://github.com/tacticians-academy/teamfight-simulator/blob/main/src/game/loop.ts?ts=2#L32). We use `requestAnimationFrame` to wait until the next draw frame before continuing the simulation. But `runLoop` has an optional `unanimated` parameter that will skip this wait.

`runLoop` is called upon clicking the start button over in [src/components/Sidebar/Sidebar.vue](https://github.com/tacticians-academy/teamfight-simulator/blob/main/src/components/Sidebar/Sidebar.vue?ts=2#L26). So we can change the call to `runLoop(performance.now(), true)` to run as fast as possible.

...Except. This now creates an infinite recursion in `runLoop` calling `requestNextFrame` calling `runLoop` calling... So if the fight goes long enough, we'll exceed the max call stack size and crash. There are also certain things that might break when run this way (in general, the sim is designed with the intent to support this, but testing is needed for edge cases). So you should run fights both ways and make sure the results are effectively the same (note that RNG will differ between individual runs).

TL;DR: If you make a few changes to the code, you can do this.

My intention is to eventually build out some basic features to allow you to automate running a sim hundreds of times, changing one variable, continuing, and aggregating the results. But as things are right now, some things will break, and regardless the simulations won't be accurate enough to provide meaningful data. But please share what results you get, good or bad, thanks!

**Comment by ky-is** [comment]
hey @jiaqp, I've made some substantial improvements to unanimated mode in 4691153. Unanimated mode will be exposed eventually through tools for statistical analysis, but I don't have an eta for that, so I'm closing this for now.

If you have any further questions or need help getting something working, please let me know!
